src/OpenApi/src/Services/OpenApiComponentService.cs

```diff
@@ -42,7 +42,26 @@ private readonly ConcurrentDictionary<(Type, ParameterInfo?), JsonObject> _schem
     {
         OnSchemaGenerated = (context, schema) =>
         {
-            schema.ApplyPrimitiveTypesAndFormats(context.TypeInfo.Type);
+            var type = context.TypeInfo.Type;
+            // Fix up schemas generated for IFormFile, IFormFileCollection, Stream, and PipeReader
+            // that appear as properties within complex types.
```

Comment:
I think the goal here is to take a type like `class Todo { public Stream Stream { get; set; } }` and update the schema for the `Stream` property.  Assuming so, does/should this act recursively?

Reply:
Ish. The underlying schema generator will use `$ref` to process the recursive type so the schema generated at the top-level will work for this. We don't handle ref IDs properly at the moment (follow-up work incoming) so the schemas are only partially correct. I'll add a test to reflect the current state of things.